### ISSUE TITLE
Profile properties display and edit improvements ideas.

### ISSUE THREAD

**Issue by Kaik** [zikula-modules/Profile#116]
@Guite ok now I see why I had to disable that listener. So the problem is that all those fields are shown both in admin view user edit and on registration form... and the reason I didn't want them to appear there is because translation didn't work.
I use old 2x Zikula old profile module (2018) I have just checked and looks like you fixed it in nov 2018 here d1e0950

ok, so... hmm it is all messed up... imho Despite the fact that translation works now.

Both views should be separated
There should be, a way to hide all those fields from registration form anyway... or even chose fields that are displayed and where...
I think hooks are better for this*
*hooks with hook settings
I'm going to write here how I see it, maybe someday I will do PR like this it is just an idea. (I might need to copy it to a separate issue and close this one let me know)
So.. since I removed all those fields from registration form I do like how clean it is, and I will not go back.

Actually now I think that registration form should be as clean as possible and additional data can be collected later, after login, similarly as it is done on facebook or other platforms. So small decentralised questions/links(ajax components etc) like "Update your age here ", Change avatar (somewhere next to avatar), Which city you are living?... could be in blocks or something. Click, update one bit, and that is it...
of course somewhere there full profile edit page as well as admin profile edit page (the way it looks/work on the image above is ok)

I know it is feature I do not expect anyone to code above or something it's just a loose concept.

I see it as hook provider (the problem with hooks is that there is no common settings practice/support and hooks providers should have one).
Profile edit hook provider
3 areas
profile edit area
admin user edit area
registration form area
So basically the only way those fields are editable is via hook provider no direct edit...

Apart from that in that profile edit hook provider in settings, admin can chose fields that are displayed.

So when admin does not want these fields in registration form then hook provider is not attached. If he wants then he can add all of the fields there or chose which he wants to display. Same for other areas. I guess in admin it is better to chose to show all fields anyway, but in edit your profile form in user account not necessarily...

Like I said it is just idea... I think I should split it to a new ticked... let me know if you want this as a new issue and what you think.

**Comment by Kaik** [comment]
So in #115  @Guite replied  


> It **is** a hook provider.
> Yes, please put the feature request into a new, clean ticket.
> 
> An easy first step would be to add a dropdown checklist to properties allowing to choose none, one or multiple from:
> 
> * Shown in admin area
> * Shown in registration
> * Shown in profile display
> * Shown in profile editing

**Comment by Kaik** [comment]
@Guite well not really I guess. I mean I'm aware that hooks are based on events, but... these listeners, are "hardcoded" there is no option to turn it on and off ...again ;) (via web interface), apart from that, hooks can be turned on and off for multiple areas independently and that ability with independent set of settings for binding is a very powerful system.

I have been thinking about adding on/off to properties, but that's one dimensional. You proposal adds a bit of room there (dropdown), but when other module will use these properties somewhere else you will have to change ProfileModule code... add this place to that dropdown. 

Since this setting is "binding" dependant it should be in binding configuration.

The way I imagine it is to have, a hook provider, and when attached to area it displays settings form. On that setting form is list of properties and additional settings for them for that particular binding.

This way I can create a module that displays let say users birthdays or cities they live in and then attach just this one property edit to that area module area... (Of course I assume some someday) some of those properties will be ajaxified :)

I know this is not 1. "easy" 2. needed now and probably complicated and depends on core feature that does not exists yet... but still this is now I see it...

I might do a PR that will move those listeners to hooks... now I'm just laying out my evil plan :)
